### PR TITLE
bpo-30757 pyinstaller added to docs, py2exe ref updated

### PR DESCRIPTION
**Pull request by den-run-ai**
It is not clear why this FAQ item is written in addition to this document:

https://github.com/python/cpython/blob/909a6f626ff343937cd3f06fda996870e7890724/Doc/faq/programming.rst#how-can-i-create-a-stand-alone-binary-from-a-python-script

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](http://bugs.python.org/) (b.p.o) to verify you have signed the CLA. This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://cpython-devguide.readthedocs.io/pullrequest.html#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

File: Doc/faq/windows.rst [hunk 1]
```diff
@@ -170,11 +170,12 @@ offender.
 How do I make an executable from a Python script?
 -------------------------------------------------
 
-See http://cx-freeze.sourceforge.net/ for a distutils extension that allows you
-to create console and GUI executables from Python code.
-`py2exe <http://www.py2exe.org/>`_, the most popular extension for building
-Python 2.x-based executables, does not yet support Python 3 but a version that
-does is in development.
+See http://cx-freeze.sourceforge.net/ and http://www.pyinstaller.org/ for the 
+distutils extensions that allows you to create console and GUI executables
```
Comment (refi64) [inline Doc/faq/windows.rst]: PyInstaller isn't a distutils extension (and AFAIK doesn't even have anything to do with distutils).

**Comment by berkerpeksag** [comment]
Thanks for the PR, but please open an issue on bugs.python.org. I agree that those two sections can be merged into one.

**Comment by den-run-ai** [comment]
added issue: http://bugs.python.org/issue30757

File: Doc/faq/windows.rst [hunk 2]
```diff
@@ -170,11 +170,13 @@ offender.
...
-`py2exe <http://www.py2exe.org/>`_, the most popular extension for building
-Python 2.x-based executables, does not yet support Python 3 but a version that
-does is in development.
+See http://cx-freeze.sourceforge.net/ for the distutils extension and
+http://www.pyinstaller.org/ for another package,
+that both allow you to create console and GUI executables
+from Python code.
+`py2exe <http://www.py2exe.org/>`_, popular extension for building
+Python 2.x-based executables, partially supports Python 3, see
+https://pypi.python.org/pypi/py2exe/.
```
Comment (briancurtin) [inline Doc/faq/windows.rst]: All three of these do the same or similar things but the grouping of the sentences reads oddly. Can we just simplify this to be more of a list? Something like...

```
The following packages can help with the creation of cross-platform console and GUI executables:
* http://cx-freeze.sourceforge.net/
* http://www.pyinstaller.org/

Additionally, http://www.py2exe.org/ exclusively supports Windows executables.
```

**Review by briancurtin — CHANGES_REQUESTED** [review]

**Comment by den-run-ai** [comment]
@briancurtin @berkerpeksag please review

File: Doc/faq/windows.rst [hunk 3]
```diff
@@ -170,11 +170,12 @@ offender.
 How do I make an executable from a Python script?
 -------------------------------------------------
 
-See http://cx-freeze.sourceforge.net/ for a distutils extension that allows you
-to create console and GUI executables from Python code.
-`py2exe <http://www.py2exe.org/>`_, the most popular extension for building
-Python 2.x-based executables, does not yet support Python 3 but a version that
-does is in development.
+The following 2 packages cx_Freeze and PyInstaller allow
```
Comment (berkerpeksag) [inline Doc/faq/windows.rst:173]: I liked Brian's suggested wording more:

> The following packages can help with the creation of console and GUI executables:

File: Doc/faq/windows.rst [hunk 4]
```diff
@@ -170,11 +170,12 @@ offender.
...
+the creation of cross-platform stand-alone console and GUI executables:
+* http://cx-freeze.sourceforge.net/
```
Comment (berkerpeksag) [inline Doc/faq/windows.rst:175]:
```rst
`cx_Freeze <http://cx-freeze.sourceforge.net/>`_
```

File: Doc/faq/windows.rst [hunk 5]
```diff
@@ -170,11 +170,12 @@ offender.
...
+* http://www.pyinstaller.org/
```
Comment (berkerpeksag) [inline Doc/faq/windows.rst:176]: PyInstaller can run on non-Windows platforms too. I think that the best we could do

* Move the content of this section to end of the "How can I create a stand-alone binary from a Python script?" section in `Doc/faq/programming.rst`
* In `Doc/faq/programming.rst`, change lines 75 and 76 to

  ```rst
  .. _faq-create-standalone-binary:

  How can I create a stand-alone binary from a Python script?
  -----------------------------------------------------------
  ```
* Remove

  ```rst
  Additionally, http://www.py2exe.org/ exclusively supports Windows
  executables.
  ```

  and add py2exe to the list:

  ```rst
  The following packages can help with the creation of console and GUI
  executables:

  * `cx_Freeze <http://cx-freeze.sourceforge.net/>`_ (Cross-platform)
  * `PyInstaller <http://www.pyinstaller.org/>`_ (Cross-platform)
  * `py2exe <http://www.py2exe.org/>`_ (Windows only)
  ```
* Then make the "How do I make an executable from a Python script?" section in this file a pointer to the "How can I create a stand-alone binary from a Python script?" section in `Doc/faq/programming.rst`:

  ```rst
  How do I make an executable from a Python script?
  -------------------------------------------------

  See :ref:`faq-create-standalone-binary` for a list of tools that can be
  used to make executables.
  ```

File: Doc/faq/windows.rst [hunk 6]
```diff
@@ -170,11 +170,12 @@ offender.
...
+
+Additionally, http://www.py2exe.org/ exclusively supports Windows executables.
```
Comment (berkerpeksag) [inline Doc/faq/windows.rst:178]:
```rst
`py2exe <http://www.py2exe.org/>`_
```

**Review by berkerpeksag — CHANGES_REQUESTED** [review]

**Comment by brettcannon** [comment]
To try and help move older pull requests forward, we are going through and backfilling 'awaiting' labels on pull requests that are lacking the label. Based on the current reviews, the best we can tell in an automated fashion is that a core developer requested changes to be made to this pull request.

If/when the requested changes have been made, please leave a comment that says, `I have made the requested changes; please review again`. That will trigger a bot to flag this pull request as ready for a follow-up review.

**Comment by csabella** [comment]
@denfromufa, please address the code reviews and resolve the merge conflicts.  Thank you!

**Comment by erlend-aasland** [comment]
Due to the lack of response (pinged both here and on bpo), I've applied @berkerpeksag and @briancurtin's suggestions and opened #26309. Closing this in favour of the new PR.

**Pull request closed without merging**